pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -615,6 +627,37 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    // NOTE: if there were no tasks to call stop we need to call stop by ourselves first
+    pub(crate) async fn persist_index_part_with_deleted_flag(
+        self: &Arc<Self>,
+    ) -> anyhow::Result<()> {
+        let index_part = {
+            let mut locked = self.upload_queue.lock().unwrap();
+
+            // We must be in stopped state because otherwise
+            // we can have inprogress index part upload that can overwrite the file
+            // with missing is_deleted flag that we going to set below
+            let stopped = match &mut *locked {
+                UploadQueue::Uninitialized => anyhow::bail!("is not Stopped but Uninitialized"),
+                UploadQueue::Initialized(_) => anyhow::bail!("is not Stopped but Initialized"),
+                UploadQueue::Stopped(stopped) => stopped,
+            };
+
+            stopped.index_part.is_deleted = true;
+
+            stopped.index_part.clone()
```

Comment:
Yeah, thats just a safety measure so if any other code faces queue in stopped state it can see the deleted mark. Not sure if it really helps, and there shouldnt be anything because the only place where this is set is delete handler, but anyway decided to do it this way. If we return the IndexPart somewhere in api it can also show the deleted value